### PR TITLE
Bring in an initial version of the FIT spec

### PR DESCRIPTION
**Pull request by sjg20**
Add the build infrastructure from devicetree.org since we want this spec to be familiar to users of that spec.

Bring in some of the source from the U-Boot tree, with a small number of updates to the initial paragraphs.

The definitions do not show as intended using this style. That can be adjusted later.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,139 @@
...
+
+The latest release of the specification can be found at http://devicetree.org/specifications/ or https://github.com/devicetree-org/devicetree-specification/releases
+
+This [repository](https://github.com/devicetree-org/devicetree-specification) holds the source for the generation of the Devicetree Specification using Sphinx and LaTeX. 
+
+## Mailing list: devicetree-spec@vger.kernel.org ##
+   * Use this mailing list for submitting patches, questions and general discussion
+   * Sign up to the mailing list at http://vger.kernel.org/vger-lists.html#devicetree-spec
+
+## Build Instructions [![ci](https://github.com/devicetree-org/devicetree-specification/actions/workflows/ci.yml/badge.svg)](https://github.com/devicetree-org/devicetree-specification/actions/workflows/ci.yml) ##
```
Comment (LeanSheng) [inline README.md:11]: This part still refers to DT spec. Is this intended for now and will be updated later?

File: NOTICE [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+Devicetree Specification
```
Comment (LeanSheng) [inline NOTICE:2]: Same as this part, is this also planned as the update for the next patch?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+	@echo "  latexdiff   to make LaTeX files including changebars against previous release"
+
+.PHONY: all help latexdiff Makefile
+
+latexdiff: latex
+	@echo "Generating LaTeX changebars..."
+	$(LATEXDIFF) --type=UNDERLINE --config VERBATIMENV=sphinxVerbatim $(BUILDDIR)/latex-previous/devicetree-specification.tex $(BUILDDIR)/latex/devicetree-specification.tex > $(BUILDDIR)/latex/devicetree-specification-changebars.tex
```
Comment (LeanSheng) [inline Makefile:24]: Still has reference to DT.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,139 @@
...
+
+## Contributions ##
+Please submit all patches to the mailing list devicetree-spec@vger.kernel.org.
+Contributions to the Devicetree Specification are managed by the gatekeepers,
+Grant Likely <grant.likely@secretlab.ca> and Rob Herring <robh@kernel.org>
+
+Anyone can contribute to the Devicetree Specification. Contributions to this project should conform 
+to the `Developer Certificate of Origin` as defined at http://elinux.org/Developer_Certificate_Of_Origin. 
+Commits to this project need to contain the following line to indicate the submitter accepts the DCO:
+```
```
Comment (LeanSheng) [inline README.md:96]: same here.

**Review by LeanSheng — COMMENTED** [review]
Just some parts are still refers to DT spec, other than that it looks good to me :)

File: README.md (hunk 1 above)
Comment (sjg20) [reply README.md:11]: fixed

File: NOTICE (hunk 2 above)
Comment (sjg20) [reply NOTICE:2]: fixed. I need to sort out the license though

File: README.md [hunk 5]
```diff
@@ -1,2 +1,140 @@
-# flat-image-tree
-Spec for the Flat Image Tree (FIT) specification
+# Document master project for the Devicetree Specification #
```
Comment (LeanSheng) [inline README.md:1]: There is still reference to DT here.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,140 @@
...
+
+The latest release of the specification can be found at http://devicetree.org/specifications/ or https://github.com/devicetree-org/devicetree-specification/releases
+
+This [repository](https://github.com/devicetree-org/devicetree-specification) holds the source for the generation of the Devicetree Specification using Sphinx and LaTeX. 
+
```
Comment (LeanSheng) [inline README.md:6]: let me check with @walterchris if we can set up a mailing list on this under OSFF.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,140 @@
...
+## Mailing list: devicetree-spec@vger.kernel.org ##
+   * Use this mailing list for submitting patches, questions and general discussion
+   * Sign up to the mailing list at http://vger.kernel.org/vger-lists.html#devicetree-spec
```
Comment (LeanSheng) [inline README.md:9]: let me check with @walterchris  if we can set up a mailing list for FIT under OSFF.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,140 @@
-# flat-image-tree
-Spec for the Flat Image Tree (FIT) specification
+# Document master project for the Devicetree Specification #
+
+The latest release of the specification can be found at http://devicetree.org/specifications/ or https://github.com/devicetree-org/devicetree-specification/releases
+
+This [repository](https://github.com/devicetree-org/devicetree-specification) holds the source for the generation of the Devicetree Specification using Sphinx and LaTeX.
```
[the remark this answers is not in the document]
Comment (sjg20) [reply README.md:5]: fixed

File: README.md (hunk 7 above)
Comment (sjg20) [reply README.md:9]: ok

Comment (LeanSheng) [reply README.md]: Hi Simon, will this works for you? :)
flat-image-tree@lists.osfw.foundation

Comment (LeanSheng) [reply README.md]: it is easy to subscribe:
https://lists.osfw.foundation/g/flat-image-tree

File: source/conf.py [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+# a list of builtin themes.
+#
+html_theme = 'alabaster'
+
+# Theme options are theme-specific and customize the look and feel of a theme
+# further.  For a list of options available for each theme, see the
+# documentation.
+html_theme_options = {
+        'github_user': 'devicetree-org',
+        'github_repo': 'devicetree-specification',
```
[the remark this answers is not in the document]
Comment (sjg20) [reply source/conf.py:114]: done

File: README.md (hunk 7 above)
Comment (sjg20) [reply README.md]: Hmmm can we get rid of the elephants?

File: source/license.rst [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+WARRANTY OF NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A
+PARTICULAR PURPOSE. IN NO EVENT SHALL LINARO OR ANY MEMBER OF
+LINARO BE LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL, EXEMPLARY,
+PUNITIVE, OR CONSEQUENTIAL DAMAGES, INCLUDING, WITHOUT LIMITATION, LOST
+PROFITS, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
+
+Questions pertaining to this document, or the terms or conditions of its
+provision, should be addressed to:
+
+| 9 Elements
```
Comment (walterchris) [inline source/license.rst:23]: it's "9elements"

Comment (sjg20) [reply source/license.rst]: ok done

File: source/references.rst [hunk 11]
```diff
@@ -0,0 +1,6 @@
+.. SPDX-License-Identifier: Apache-2.0
+
+References
+==========
+
+.. [dtspec]
```
Comment (LeanSheng) [inline source/references.rst:6]: do you want to add the dtspec link here?
https://www.devicetree.org/specifications/

File: README.md (hunk 7 above)
Comment (LeanSheng) [reply README.md]: I replace the 'elephants' with the FIT logo.

Comment (sjg20) [reply README.md]: thanks!

File: source/references.rst (hunk 11 above)
Comment (sjg20) [reply source/references.rst]: added

**Pull request merged**